server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -140,7 +140,6 @@ static void executeQuery(SearchContext searchContext) throws QueryPhaseExecution
 
         RescorePhase.execute(searchContext);
         SuggestPhase.execute(searchContext);
-        AggregationPhase.execute(searchContext);
```

Comment:
that's an interesting change. Do we effectively still need an aggregation phase? The preProcess is the only bit that is called at this point?

Reply:
If we go this way, we will probably not needed it, but I am not sure we can do it this way, let me explain:

One thing I notice is that this phase `AggregationPhase#execute` is run after we [remove the timeouts](https://github.com/elastic/elasticsearch/blob/fd5066ae52e093283b7dd8e0eac1131653611d17/server/src/main/java/org/elasticsearch/search/query/QueryPhase.java#L257) from the IndexSearcher. If we want the same behaviour here, we will need to make sure we call reduce after we remove the timeouts. If we don't do that the postCollection method can throw Timeouts.

This will need to be made inside the `ContextIndexSearcher` and it will be even more tricky when we move to Lucene timeouts. I think that as far as we don't move postCollections to the slice thread, we will not be able to use the reduce method.